src/java.desktop/windows/native/libawt/windows/ThemeReader.cpp

```diff
@@ -116,8 +115,8 @@ BOOL InitThemes() {
     DTRACE_PRINTLN1("InitThemes hModThemes = %x\n", hModThemes);
     if(hModThemes) {
         DTRACE_PRINTLN("Loaded UxTheme.dll\n");
-        OpenThemeDataFunc = (PFNOPENTHEMEDATA)GetProcAddress(hModThemes,
-                                                        "OpenThemeData");
+        OpenThemeDataFuncForDpi = (PFNOPENTHEMEDATAFORDPI)GetProcAddress(
+                                   hModThemes, "OpenThemeDataForDpi");
```

Comment:
Do we plan to backport this change to the old versions of JDK? If yes will we plan to support old WIndows? If yes probably we should fallback to the "OpenThemeData" if "OpenThemeDataForDpi" is not available?

Reply:
I guess this could (should?) be backported but Windows 10 is the oldest supported OS no matter what the JDK version.
And only the more recent versions of Windows 10 at that ...

Reply:
> I guess this could (should?) be backported but Windows 10 is the oldest supported OS no matter what the JDK version. And only the more recent versions of Windows 10 at that ...

I think it should be backported. The `OpenThemeDataForDpi` function was introduced in Windows 10 version 1703 which is already 5 years old and is unsupported, that is the function is available in the supported versions of Windows 10 and 11.

I may be useful to be able to run Java on previous versions though, yet it's not a priority.

Reply:
> It may be useful to be able to run Java on previous versions though, yet it's not a priority.

Reply:
> > It may be useful to be able to run Java on previous versions though, yet it's not a priority.
> 
> Then probably it is better to add a fallback since that method was added to the win 10 after GA, and it does not exist in the early version.

But the earlier versions of Windows 10 are already unsupported, including the one where the `OpenThemeDataForDpi` function was added.

If required, the fallback could be implemented under a new bug. Anyway, I don't have a strong opinion on this matter.

Reply:
I think it's *unfeasible* to provide the fallback to keep the code simpler.

If `OpenThemeDataForDpi` isn't available, we cannot get per-DPI images. Then we have to fallback to using the DPI of the main display and scale up / down all the returned values and images.